specification/dartLangSpec.tex

```diff
@@ -3411,89 +3414,54 @@ \subsubsection{Constant Constructors}
 <constantConstructorSignature> ::= \CONST{} <qualified> <formalParameterList>
 \end{grammar}
 
-%\commentary{Spell out subtleties: a constant constructor call within the initializer of a constant constructor is treated as a ordinary constructor call (a new), because the arguments cannot be assumed constant anymore. In practice, this means two versions are compiled and analyzed. One for new and one for const.}
-
-\commentary{
-All the work of a constant constructor must be handled via its initializers.
+\commentary{%
+Constant constructors have stronger constraints than other constructors.
+For instance, all the work of a non-redirecting generative constant constructor
+must be done in its initializers
+and in the initializing expressions of
```

Comment:
Not even in the initializing expressions of its instance variables - those must all be compile-time constant, so there is no work there.

Reply:
Constant expression evaluation is allowed to take place late. Added a few words clarify this.